astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Set the time coordinate state of a FITS Binary Table.
+
+        Parameters
+        ----------
+        hdr : `~astropy.io.fits.header.Header`
+            FITS Header
+        table : astropy.table.Table
+            The table whose time columns are to be read as Time
+        """
+        for key, value, comment in hdr.cards:
+            if (key.upper() in TIME_KEYWORDS):
+
+                self.set_global_time(key, value, comment)
+                hdr.remove(key)
+
+            elif (is_time_column_keyword(key.upper())):
+
+                self.set_column_override(key, value, comment)
+                hdr.remove(key)
+
+        self.convert_time_columns(table)
+
+    @classmethod
+    def replace_time_table(cls, table):
+        """
+        Replace Time columns in a Table with non-mixin columns containing
+        each element as a vector of two doubles (jd1, jd2) and return a FITS 
+        header with appropriate time coordinate keywords.
+        jd = jd1 + jd2 represents time in the Julian Date format with 
+        high-precision.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The table whose Time columns are to be replaced.
+
+        Returns
+        -------
+        table : astropy.table.Table
+            The table with replaced Time columns
+        hdr : `~astropy.io.fits.header.Header`
+            Header containing Cards associated with the FITS time coordinate
+        """
+        # Shallow copy of the input table
+        newtable = table.copy(copy_data=False)
+
+        # AstroPy FITS File specification
+        hdr = Header([Card(key='COMMENT', value=val) for val in cls.ASTROPY_SPECIFIER])
+
+        # Global time coordinate frame keywords
+        hdr.extend([Card(keyword=key, value=val[0], comment=val[1]) 
+                     for key, val in cls.GLOBAL_TIME_INFO.items()])
+
+        time_cols = table.columns.isinstance(Time)
+
+        # Geocentric Position
+        pos_geo = None
+
+        for col in time_cols:
+            jd12 = np.array([col.jd1, col.jd2])
+            jd12 = np.rollaxis(jd12, 0, jd12.ndim)
+            newtable.replace_column(col.info.name, Column(jd12, unit='d'))
+
+            # Get column position(index)
+            n = table.colnames.index(col.info.name) + 1
+
+            # Time column override keywords
+            hdr.append(Card(keyword='TCTYP%d' %n, value=col.scale.upper()))
+
+            # Time column reference positions
+            if col.location is None:
+                if pos_geo is not None:
+                    warnings.warn(
+                        'Time Column "{0}" has no specified location, but global Time Position '
+                        'is present, which will be the default for this column in '
+                        'FITS specification.'.format(col.info.name),
+                        AstropyUserWarning)
+            else:
+                hdr.append(Card(keyword='TRPOS%d' %n, value='TOPOCENTER'))
```

Comment:
> killer feature

I think it will be quite a while before Python 3.6 is the minimal required version... :sweat_smile: